crypt-des-obsolete.c

```diff
@@ -116,17 +124,26 @@ do_setkey_r (const char *key, struct des_ctx *ctx)
 #endif
 
 #if INCLUDE_setkey_r
+#if ENABLE_OBSOLETE_API_ENOSYS
+void
+setkey_r (ARG_UNUSED (const char *key), ARG_UNUSED (struct crypt_data *data))
+{
+  /* Pretend this function is not supported.  */
+  errno = ENOSYS;
+}
+#else
```

Comment:
It's fine to put `ARG_UNUSED` on an argument that _may_ be used, and I don't like duplicating an entire function definition in two branches of a preprocessor conditional, I worry that the prototypes will get out of sync.  Also, in this configuration, we're not _pretending_ the function isn't supported.  Suggest instead

``` c
#if INCLUDE_setkey_r
void
setkey_r (ARG_UNUSED (const char *key), ARG_UNUSED (struct crypt_data *data))
{
#if ENABLE_OBSOLETE_API_ENOSYS
  /* This function is not supported in this configuration.  */
  errno = ENOSYS;
#else
  do_setkey_r (key, get_des_ctx (data));
#endif
}
SYMVER_setkey_r;
#endif
```

Reply:
Fixed.